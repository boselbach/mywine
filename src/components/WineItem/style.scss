.wine-item {
  border-bottom: 1px solid black;
  background: #722f37;
  color: #fff;
  min-height: 50px;
  cursor: pointer;
  position: relative;

  header {
    display: flex;
    flex-direction: row;
    padding-top: 10px;

    p {
      font-size: 30px;
      text-indent: 20px;
    }
  }

  .status {
    position: absolute;
    right: 0;
    height: 30px;
    width: 30px;
    top: 20px;
    font-size: 20px;
  }

  .details {
    display: none;
    padding: 20px;

    &.active {
      display: block;
    }

    .item {
      display: flex;
      flex-direction: row;
      margin-bottom: 10px;

      p {
        flex: 1;

        &:first-child {
          flex: 0 150px;
        }
      }
    }

    .description {
      padding: 20px 0;
    }

    .controls {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 40px;

      button {
        border-radius: 0;
        border: 1px solid #000;
        padding: 5px;
        color: #fff;
        background-color: #000;
        margin-right: 5px; 
      }
    }
  }
}