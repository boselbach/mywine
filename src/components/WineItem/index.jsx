import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {} from './style.scss';

class WineItem extends PureComponent {
  static propTypes = {
    item: PropTypes.shape({
      name: PropTypes.string,
      country: PropTypes.string,
      region: PropTypes.string,
      grapes: PropTypes.string,
      producer: PropTypes.string
    })
  }

  static defaultProps = {
    name: '',
    country: '',
    region: '',
    description: ''
  }

  state = {
    active: false
  }

  handleClick = () => {
    this.setState(prevState => ({active: !prevState.active}));
  }

  render() {
    let detailsClass= ['details'];
    const { active } = this.state;
    const {
      item: {
        id,
        name,
        country,
        region,
        price,
        description
      }
    } = this.props;
    let statusText = active ? '-' : '+';
    let statusTitle = active ? 'Close' : 'Open';

    if (active) {
      detailsClass.push('active');
    }

    return (
      <div onClick={this.handleClick} className="wine-item">
        <header>
          <p>{name}</p>
        </header>
        
        <div className={detailsClass.join(' ')}>
          <div className="content">
            <div className="item">
              <p>Country:</p>
              <p>{country}</p>
            </div>
          
            <div className="item">
              <p>Region:</p>
              <p>{region}</p>
            </div>

            <div className="item">
              <p>Price:</p>
              <p>{price}</p>
            </div>

            <div className="item">
              <p>Description:</p>
              <p>{description}</p>
            </div>
          
          </div>
          <div className="controls">          
            <button className="remove" onClick={() => this.props.remove(id)}>Remove</button>
          </div>
        </div>
        <div className="status" title={statusTitle}>{statusText}</div>
      </div>
    )
  }
}

export default WineItem;