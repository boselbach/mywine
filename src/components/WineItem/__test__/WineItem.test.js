import React from 'react';
import { shallow } from 'enzyme';
import WineItem from '../';

describe('WineItem', () => {
  it('renders correctly', () => {
    const component = shallow(
      <WineItem />
    );

    expect(component).toMatchSnapshot();
  });
});
