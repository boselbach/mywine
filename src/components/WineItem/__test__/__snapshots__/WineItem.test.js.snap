// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WineItem renders correctly 1`] = `
<div>
  wineItem
</div>
`;
